### ISSUE TITLE
Better AndroidTestParser implementation

### ISSUE THREAD

**Issue by tagantroy** [MarathonLabs/marathon#480]
**Is your feature request related to a problem? Please describe.**
Current implementation of AndroidTestParser relies on DexParser, it is not possible to handle parameterized tests properly. Marathon cannot distribute parameterized tests between workers properly. 

**Describe the solution you'd like**
Switch to `am instrument -e log true`.

**Comment by trevjonez** [comment]
also seen issues with nested tests. roughly this:

```kotlin
@RunWith(...)
abstract class SomeTestPatternBaseclass {

 //Rules.....
 
   class SpecificScenarios: SomeTestPatternBaseclass() {
      @Test
      fun fooTest() {...}
    }
}
```

**Comment by Malinskiy** [comment]
Closed with #539
